src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -0,0 +1,868 @@
+     */
+    public ByteBuffer asIndexBytes(ByteBuffer value)
+    {
+        if (value == null)
+            return null;
+
+        if (isInetAddress())
+            return encodeInetAddress(value);
+        else if (isBigInteger())
+            return encodeBigInteger(value);
+        else if (isBigDecimal())
+            return encodeDecimal(value);
+        return value;
+    }
+
+    public float[] decomposeVector(ByteBuffer byteBuffer)
+    {
+        assert isVector();
+        return ((VectorType<?>) indexType).composeAsFloat(byteBuffer);
+    }
+
+    public boolean supports(Operator operator)
+    {
+        if (operator == Operator.LIKE ||
+            operator == Operator.LIKE_CONTAINS ||
+            operator == Operator.LIKE_PREFIX ||
+            operator == Operator.LIKE_MATCHES ||
+            operator == Operator.LIKE_SUFFIX) return false;
+
+        // ANN is only supported against vectors, and vector indexes only support ANN
+        if (operator == Operator.ANN)
+            return isVector();
+
+        Expression.IndexOperator indexOperator = Expression.IndexOperator.valueOf(operator);
+
+        if (isNonFrozenCollection())
+        {
+            if (indexTargetType == IndexTarget.Type.KEYS) return indexOperator == Expression.IndexOperator.CONTAINS_KEY;
+            if (indexTargetType == IndexTarget.Type.VALUES) return indexOperator == Expression.IndexOperator.CONTAINS_VALUE;
+            return indexTargetType == IndexTarget.Type.KEYS_AND_VALUES && indexOperator == Expression.IndexOperator.EQ;
+        }
+
+        if (indexTargetType == IndexTarget.Type.FULL)
+            return indexOperator == Expression.IndexOperator.EQ;
+
+        if (indexOperator != Expression.IndexOperator.EQ && EQ_ONLY_TYPES.contains(indexType)) return false;
+
+        // RANGE only applicable to non-literal indexes
+        return (indexOperator != null) && !(isLiteral() && indexOperator == Expression.IndexOperator.RANGE);
+    }
+
+    @Override
+    public String toString()
+    {
+        return MoreObjects.toStringHelper(this)
+                          .add("column", columnMetadata)
+                          .add("type", indexType)
+                          .add("indexType", indexTargetType)
+                          .toString();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (obj == this)
+            return true;
+
+        if (!(obj instanceof IndexTermType))
+            return false;
+
+        IndexTermType other = (IndexTermType) obj;
+
+        return Objects.equals(columnMetadata, other.columnMetadata) && (indexTargetType == other.indexTargetType);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(columnMetadata, indexTargetType);
+    }
+
+    private EnumSet<Capability> calculateIdentityBitmap(ColumnMetadata columnMetadata, List<ColumnMetadata> partitionKeyColumns, IndexTarget.Type indexTargetType)
```

Comment:
After having changed this to an `EnumSet`, the method could be simply named `calculateCapabilities`.